docs/developer/release-process.md

```diff
@@ -61,10 +61,11 @@ To create a new release, follow these steps:
 10. Prepare and merge a PR into the main branch to update the [README](/README.md) to include the information about
     the latest release and also the [changelog](/CHANGELOG.md). Also update any installation instructions to ensure
     that the supported Gateway API and NGF versions are correct. Specifically, helm README and `site/content/includes/installation/install-gateway-api-resources.md`.
-11. Close the issue created in Step 1.
-12. Ensure that the [associated milestone](https://github.com/nginxinc/nginx-gateway-fabric/milestones) is closed.
-13. Verify that published artifacts in the release can be installed properly.
-14. Submit the `conformance-profile.yaml` artifact from the release to the [Gateway API repo](https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports).
+11. Update the production branch to the latest release branch in [Netlify](https://app.netlify.com/sites/nginx-gateway-fabric/configuration/deploys#branches-and-deploy-contexts). Log in using Gitlab SSO and ask the docs team for permission to access the NGF project.
```

Comment:
It's possible. For other documentation sets automatic deploy is disabled for this reason: it's merged into the codebase but doesn't go live (Through deploying to the production stage) until human intervention occurs.

The final sentence of step 11 seems slightly out of sequence since getting permissions should only need to happen once on a per-user basis.

Reply:
How extensive would "testing the docs" be though? The release dev goes through every page and link to see if it works? That's not very scalable.

I can mention that the dev who updates the netlify branch should ensure that its deployment pipeline passes. If it doesn't, then we address it. But I wouldn't want that to block a release.

Reply:
They are, but it shouldn't prevent us from publishing images and artifacts is what I mean.